components/script_bindings/conversions.rs

```diff
@@ -31,10 +31,17 @@ use crate::inheritance::Castable;
 use crate::num::Finite;
 use crate::reflector::{DomObject, Reflector};
 use crate::root::DomRoot;
+use crate::script_runtime::JSContext as SafeJSContext;
 use crate::str::{ByteString, DOMString, USVString};
 use crate::trace::RootedTraceableBox;
 use crate::utils::{DOMClass, DOMJSClass};
 
+/// A safe wrapper for `ToJSValConvertible`.
+#[allow(unsafe_code)]
+pub fn safe_to_jsval<T: ToJSValConvertible>(cx: SafeJSContext, val: T, rval: MutableHandleValue) {
+    unsafe { val.to_jsval(*cx, rval) };
+}
+
```

Comment:
Might be nice to make this a trait with `fn to_jsval_safe` and implement it for `T: ToJSValConvertible`. It's a bit more code, but it will make the callsites a lot nicer.

Reply:
Thanks, I'll give it a try.